src/http/h1/decode.rs

```diff
@@ -37,7 +37,7 @@ enum Kind {
     /// A Reader used when a Content-Length header is passed with a positive integer.
     Length(u64),
     /// A Reader used when Transfer-Encoding is `chunked`.
-    Chunked(Option<u64>),
+    Chunked(ChunkedState, usize),
```

Comment:
A `u64` was used before in case a chunk size claimed the incoming chunk was >4GB. On a 32-bit platform, `usize` cannot represent a number bigger than ~4 billion. Sounds absurd to send a chunk so big, but seemed prudent to be able to survive it.